twitchpopups.htm

```diff
@@ -102,24 +112,66 @@ <h1 id="popuptext"></h1>
                     $("#popuptext").text(commandName.substr(7).toUpperCase());
                     doAnimation();
                 } else if (commandName.startsWith("!delete")) {
-                  deleteAnimation();
+                    deleteAnimation();
                 } else if (commandName.startsWith("!hotseat ")) {
-                  hotSeatIsOn = true;
-                  $("#popupbox").show();
-                  hotSeatUser = commandName.substr(10).toLowerCase();
-                  $("#popupbox").css({"background-color":hotseatBg}); 
-                  $("#popuptext").text(`${hotseatEmoji}  ${hotSeatUser.toUpperCase()} IS IN THE HOTSEAT ${hotseatEmoji}`);
-                  doAnimation();
+                    hotSeatIsOn = true;
+                    $("#popupbox").show();
+                    voteTopic = commandName.substr(10);
+                    $("#popupbox").css({"background-color":hotseatBg}); 
+                    $("#popuptext").text(`${hotseatEmoji}  ${hotSeatUser.toUpperCase()}`);
+                    doAnimation();
+                } else if (commandName.startsWith("!vote ")) {
+                    // Start with the topic for your vote
+                    $("#popupbox").show();
+                    voteTopic = commandName.substr(6).toUpperCase();
+                    $("#popupbox").css({"background-color": votingBg});
+                    $("#popuptext").text(`VOTE: ${voteTopic} 🔴`);
+                    doAnimation();
+                } else if (commandName.startsWith("!voteopt ")) {
+                    // Then add your options
+                    let option = commandName.substr(9).toUpperCase();
+                    options[option] = {};
+                } else if (commandName.startsWith("!voteoptclear")) {
+                    // Made a mistake? clear and re-add your options
+                    options = [];
+                } else if (commandName.startsWith("!votestart") && !votingIsOn) {
+                    // Once options are added, start the vote!
+                    votingIsOn = true;
+                    $("#popuptext").text(`VOTE: ${voteTopic} 🟢`);
+                } else if (commandName.startsWith("!voteresults") && votingIsOn) {

```

Comment:
should you clear the options array at the end of this if block?

Reply:
You could, but leaving it open allows more flexibility to re-open the vote or add a new option and re open etc.

Reply:
Close if you agree :)

Reply:
I only really envisaged the total being announced at the end of the poll, but you're right, there is scope for updates along the way, especially with the option that people can change their mind. that could be kinda fun if the vote changed massively based on the first peek. Maybe the host doesn't like the results so far and makes it known that there's the opportunity to change your mind.